### PR TITLE
Refactor/sort out dependencies

### PR DESCRIPTION
**Pull request by THOR300**
This Pull Request: 
---
- Integrates refactoring changes to clean up the dependencies and workflow files in the repo 
- This branch is pulled from the following branch / PR: https://github.com/climatepolicyradar/navigator-data-ingest/pull/69
- Updated method of pulling documents from s3.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,21 +8,29 @@ on:
...
+  test:
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v2
+      - name: Setting up Github Actions
+        uses: actions/checkout@v3
+
+      - name: Setting up Python
+        uses: actions/setup-python@v2
```
Comment (olaughter) [inline .github/workflows/ci.yml:20]: There is a v4 for both of these

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,21 +8,29 @@ on:
...
-      - name: Build
-        run:  make build
+      - name: Install Dependencies
+        run: |
+          python -m pip install "poetry==1.3.2" && poetry install --only integration-test
 
-      - name: Run Unit Tests
-        run: echo TODO-TODO-TODO-TODO-TODO-TODO-TODO-TODO-TODO-TODO-TODO-TODO
+      - name: Building the Image
+        run: make build
```
Comment (olaughter) [inline .github/workflows/ci.yml:30]: Love to see a todo getting done!

File: HOW_TO_UPDATE_TESTS.md [hunk 3]
```diff
@@ -85,7 +85,7 @@ If you are trying to figure out what the variables are look in the env var secti
 Example:
 
 ```shell
-    docker run -e AWS_ACCESS_KEY_ID=XXX -e AWS_SECRET_ACCESS_KEY=XXX -e API_HOST="" -e MACHINE_USER_EMAIL="" -e MACHINE_USER_PASSWORD="" navigator-data-ingest-test --pipeline-bucket pipbucket123123123 --document-bucket docbucket123123123 --updates-file-name new_and_updated_documents.json --output-prefix ingest_unit_test_parser_input --embeddings-input-prefix ingest_unit_test_embeddings_input --indexer-input-prefix ingest_unit_test_indexer_input --execution-id 123456 --execution-data-prefix execution_data
+    docker run -e AWS_ACCESS_KEY_ID=XXX -e AWS_SECRET_ACCESS_KEY=XXX -e API_HOST="" -e MACHINE_USER_EMAIL="" -e MACHINE_USER_PASSWORD="" navigator-data-ingest --pipeline-bucket pipbucket123123123 --document-bucket docbucket123123123 --updates-file-name new_and_updated_documents.json --output-prefix ingest_unit_test_parser_input --embeddings-input-prefix ingest_unit_test_embeddings_input --indexer-input-prefix ingest_unit_test_indexer_input --execution-id 123456 --execution-data-prefix execution_data
```
Comment (olaughter) [inline HOW_TO_UPDATE_TESTS.md:88]: Don't worry about this now, but just flagging we might want to find a better way to handle the docker run command here, the inputs are kinda dense and maybe moving to envfiles or something would be nice...

File: integration_tests/conftest.py [hunk 4]
```diff
@@ -1,21 +1,47 @@
...
+                    for obj in response["Contents"]
+                    if obj["Key"].endswith(suffix)
+                ]
+            )
+
+        # Convert to s3 paths and return
+        return [
+            S3Path(os.path.join("s3://", bucket, file)) for file in files_with_suffix
+        ]
+    return []
```
Comment (olaughter) [inline integration_tests/conftest.py:36]: There might be a cleaner approach to getting all the objects with the paginator if your interested, this snippet would get most of the way, you might be able to repurpose that search method to search the suffix as well: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#filter-objects-by-last-modified-time-using-jmespath

File: integration_tests/conftest.py [hunk 5]
```diff
@@ -1,21 +1,47 @@
...
+PIPELINE_BUCKET = os.environ.get("INGEST_PIPELINE_BUCKET")
+S3_CLIENT = boto3.client("s3")
 
-def get_bucket_files_with_suffix(bucket: S3Path, suffix: str) -> list[S3Path]:
-    """Get all the files in a bucket with a given suffix."""
-    bucket_files = []
-    for pattern in ["*", "*/*", "*/*/*", "*/*/*/*"]:
-        files = list(bucket.glob(pattern + suffix))
-        bucket_files.extend(set(files))
-    return bucket_files
```
Comment (olaughter) [inline integration_tests/conftest.py:12]: So was this what was causing those test failures?

**Review by olaughter — APPROVED** [review]
Nice one

File: integration_tests/conftest.py (hunk 5 above)
Comment (THOR300) [reply integration_tests/conftest.py:12]: Yep for some reason the glob just wasn't working. Switching to boto3 seemed to resolve the issue. 

I don't know whether some check in the poetry deps affected cloud path lib or something.

File: HOW_TO_UPDATE_TESTS.md (hunk 3 above)
Comment (THOR300) [reply HOW_TO_UPDATE_TESTS.md:88]: Yes agreed that it's not ideal! To be fair we could export the vars and use the make command.

**Pull request merged**